src/pcrlock/pcrlock.c

```diff
@@ -4111,8 +4124,9 @@ static ssize_t event_log_calculate_component_combinations(EventLog *el) {
                 /* Overflow check */
                 if (c->n_variants > (size_t) (SSIZE_MAX/count))
                         return log_error_errno(SYNTHETIC_ERRNO(E2BIG), "Too many component combinations.");
-
-                count *= c->n_variants;
+                /* If no variant, this will lead to count being 0 and sigfpe */
+                if (c->n_variants)
```

Comment:
same here, explicit non-zero comparison please

also, we generally have this rule that in case of if blocks we prefer if the exceptional case is the indented one, and the non-exception, regular case is not indented, when possible.

hence, please conside to change this to:

```c
if (c->n_variants == 0)
        continue;
```